src/bos/operators/power_on.py

```diff
@@ -107,7 +110,17 @@ def _set_bss(self, components):
                              "nodes: {nodes}. Error: {err}")
             else:
                 token = resp.headers['bss-referral-token']
-                record_boot_artifacts(token, kernel, kernel_parameters, initrd)
+                attempts = 0
+                while attempts <= retries:
+                    try:
+                        record_boot_artifacts(token, kernel, kernel_parameters, initrd)
+                        break
+                    except Exception as err:
+                        attempts += 1
```

Comment:
Do you want to put a short pause in here in case the system needs a little time to catch up, or are 3 tries within 10ms good for failing the whole thing?

Reply:
During testing, the connection re-establishes it immediately after the initial failure during the follow-up attempt, so I don't think sleeping is needed.